### PR TITLE
detects libatomic flags in a cmake module

### PR DESCRIPTION
**Pull request by Benitoite**
Detect libatomic flags by cmake module rather than providing for their exclusions, see #6821

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -565,6 +565,12 @@ if(WITH_MYFILE_MMAP)
     add_definitions(-DMYFILE_MMAP)
 endif()
 
+# Atomic is required in some builds: #6821
+find_package(ATOMIC)
+if(ATOMIC_FOUND)
+	target_link_libraries(executable ${ATOMIC_LIBRARIES})
```
Comment (Floessie) [inline CMakeLists.txt:571]: @Benitoite `executable` wasn't meant literally. It should rather be `rawtherapee` and another line with `rawtherapee-cli` or completely different by adding `${ATOMIC_LIBRARIES}` to `EXTRA_LIB_RTGUI` and the corresponding variable for RT.

I'm surprised this works as is...

**Comment by kmilos** [comment]
FYI, see also how CMake itself does it for more ideas (although the principle is pretty much the same): https://gitlab.kitware.com/cmake/cmake/-/blob/4cd207b6f3b2b262cf4cc8eb46a2320906542d1e/Source/Checks/cm_cxx_features.cmake#L100-105

File: CMakeLists.txt (hunk 1 above)
Comment (Benitoite) [reply CMakeLists.txt:571]: I don't have a system requiring Atomic in a gnu build, so I couldn't trigger any error on that line... I think the targets we need are `rth` and `rth-cli` as set in `rtgui/CMakeLists.txt`

https://github.com/Beep6581/RawTherapee/blob/dev/rtgui/CMakeLists.txt#L330C1-L330C1

Comment (Benitoite) [reply CMakeLists.txt:571]: #6832 has the properly named targets.

**Pull request merged**